src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1184,11 +1184,22 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
                 $files[$locale][] = (string) $file;
             }
 
+            $projectDirLength = \strlen($projectDir = $container->getParameter('kernel.project_dir'));
+
             $options = array_merge(
                 $translator->getArgument(4),
                 [
                     'resource_files' => $files,
-                    'scanned_directories' => array_merge($dirs, $nonExistingDirs),
+                    'scanned_directories' => $scannedDirectories = array_merge($dirs, $nonExistingDirs),
+                    'cache_vary' => [
+                        'scanned_directories' => array_map(static function (string $dir) use ($projectDir, $projectDirLength): string {
+                            if (0 === strncmp($dir, $projectDir, $projectDirLength)) {
```

Comment:
does this work fine on Windows in all cases, or could we have path separator mismatches.

Reply:
https://github.com/symfony/symfony/pull/34397/files#diff-b0c828e732d3d31f0b2bfb2378bd1011R828 normally asserts that it works. The project dir needs to be correctly removed for the assertion to pass.